src/renderer/library/components/settings/RecoverySettings.tsx

```diff
@@ -0,0 +1,168 @@
+import { ipcRenderer } from "electron";
+import * as path from "path";
+import { _CONTINUOUS_INTEGRATION_DEPLOY, _NODE_ENV } from "readium-desktop/preprocessor-directives";
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface IBaseProps extends TranslatorProps {
+}
+// IProps may typically extend:
+// RouteComponentProps
+// ReturnType<typeof mapStateToProps>
+// ReturnType<typeof mapDispatchToProps>
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface IProps extends IBaseProps, ReturnType<typeof mapStateToProps>, ReturnType<typeof mapDispatchToProps> {
+}
+
+interface IState {
+    recovery: string[];
+    name: string;
+};
+
+class RecoverySettings extends React.Component<IProps, IState> {
+    configDataFolderPath: string;
+    recoveryFilePath: string;
+
+    constructor(props: IProps) {
+        super(props);
+        this.state = {
+            recovery: [],
+            name: "",
+        };
+        ipcRenderer.on("GETPATH", async (_0: any, userDataPath: string) => {
+
+            try {
+                this.configDataFolderPath = path.join(
+                    userDataPath,
+                    `config-data-json${(_NODE_ENV === "development" || _CONTINUOUS_INTEGRATION_DEPLOY) ? "-dev" : ""}`,
+                );
+                const RECOVERY_FILENAME = "recovery";
+                this.recoveryFilePath = path.join(
+                    this.configDataFolderPath,
+                    RECOVERY_FILENAME,
+                );
+                await this.getRecoveryState();
+
+            } catch (e) {
+                // ignore
+                console.error(e);
+
+            }
+        });
+    }
+
+    public render(): React.ReactElement<{}> {
+        // const { __ } = this.props;
+
+        return (
+            <>
+                <h3>{"Select a recovery file"}</h3>
+                <h5>{"Be careful"}</h5>
+                <form className={styles.languages_list}>
+                    {this.state.recovery.map((date, i) =>
+                        <div key={i}>
+                            <input
+                                id={"radio-" + date}
+                                type="radio"
+                                name="language"
+                                onChange={() => this.setRecoveryFile(date)}
+                                checked={this.state.name === date}
+                            />
+                            <label htmlFor={"radio-" + date}>
+                                {this.state.name === date && <SVG svg={DoneIcon} ariaHidden />}
+                                {date ? new Date(parseInt(date, 10)).toString().split("GMT")[0] : "actual state"}
+                            </label>
+                        </div>,
+                    )}
+                </form>
+            </>
+        );
+    }
+
+    private getRecoveryState = async () => {
+
+        try {
+
+            const files = await fsp.readdir(this.configDataFolderPath);
```

Comment:
May I suggest moving this to the Electron main process, and then instead of `webContents.send("GETPATH", app.getPath("userData"));`, send the data parameter of `this.setState({ recovery });` via IPC.